shuup/api/encoders.py

```diff
@@ -11,13 +11,17 @@
 from rest_framework.renderers import JSONRenderer
 from rest_framework.utils import encoders
 
+from shuup.utils.money import Money
+
 LOG = logging.getLogger(__name__)
 
 
 class ExtJSONEncoder(encoders.JSONEncoder):
     def default(self, obj):
         if isinstance(obj, Enum):
             return obj.value
+        if isinstance(obj, Money):
+            return obj.value
```

Comment:
The currency should be exported too. Maybe a money instance should be encoded as `{'value': obj.value, 'curreny': obj.currency}` ?

Reply:
good question. this sounds like a special field for me, not a job for the encoder. what do you think?

Reply:
Yeah I think valid point. And sure we want to keep the currency information.

Reply:
I guess I can create a the field to be used with `Money` objects, it will parse and serialize the object as you said and the encoder can be a fallback to just the float value.
We did this because we were having issues with a particular scenario with builtin `decimal.py`. We have a Python 3.4 compiled without decimal C module, weird though. And rebuilding/updating python at that point was not an option, so we had to "fix" our code.
The error was:
```
ValueError: could not convert string to float: '0E-9 USD'
  File "django/core/handlers/base.py", line 174, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "django/core/handlers/base.py", line 172, in get_response
    response = response.render()
  File "django/template/response.py", line 160, in render
    self.content = self.rendered_content
  File "rest_framework/response.py", line 72, in rendered_content
    ret = renderer.render(self.data, accepted_media_type, context)
  File "rest_framework/renderers.py", line 103, in render
    separators=separators
  File "json/__init__.py", line 237, in dumps
    **kw).encode(obj)
  File "json/encoder.py", line 192, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "json/encoder.py", line 250, in iterencode
    return _iterencode(o, 0)
  File "shuup/api/encoders.py", line 21, in default
    return super(ExtJSONEncoder, self).default(obj)
  File "rest_framework/utils/encoders.py", line 47, in default
    return float(obj)
  File "shuup/utils/_unitted_decimal.py", line 176, in __float__
    return super(UnittedDecimal, self).__float__(**kwargs)
  File "python3.4/decimal.py", line 1614, in __float__
    return float(s)
```

See that the code executing python code instead of the C Module. See the required version of the lib here: https://github.com/python/cpython/blob/3.4/Lib/decimal.py#L149 The server has 2.4.0. So this was breaking there.
I couldn't create a good unit test for this scenario, basically if I remove this import here https://github.com/python/cpython/blob/3.4/Lib/decimal.py#L6400 I can make tests crash :smile: So if you have a good idea, I appreciate. Now you have context @suutari-ai , I will edit the commit and PR info to include this too.

Reply:
Being able to do `float(Money(10, 'USD'))` is not a very good feature anyway. IMHO it could even be better if that wouldn't work and there was just the explicit way to get the value out of the Money object, i.e. `Money(10, 'USD').value`.

If there is serializers that expect to serialize the money object as simple float, they're simply wrong and should be fixed to either serialize them as

Reply:
yes, indeed. well, maybe `Money.__float__` should raise exception then.
I put the `MoneyField` in the PR already, it deals with the object `{"currency: "USD", "value": 10}`. To deal with Money values, I guess a `ReadOnlyField(source="order.taxful_total_price.value")` is enough.